wordpress-mobile/WordPress-iOS
# Comment Notifications: remove Trash swipe action

Fixes #18257 

This removes the comment `Trash` swipe action in the Notifications list. In addition:
- The word `Comment` has been added to the `Approve`/`Unapprove` button for clarity.
- Swipe action events are now tracked.
  - Approve/Unapprove - uses the existing `notifications_comment_(un)approved` events and adds a `notifications` or `notification_details` source.
  - Mark as read/unread - new `notification_mark_as_(un)read_tapped` events have been added.

| Before | After |
|--------|-------|
| <img width="440" alt="unapprove_before" src="https://user-images.githubusercontent.com/1816888/162852683-eac39c9b-53cf-4702-86dc-7920814d769e.png"> | <img width="441" alt="unapprove_after" src="https://user-images.githubusercontent.com/1816888/162852687-1b535d58-59a5-409d-93d8-4933a277ef42.png"> |
| <img width="440" alt="approve_before" src="https://user-images.githubusercontent.com/1816888/162852685-d7952bc0-f29d-45b1-8fef-01bbd7b5c1ad.png"> | <img width="439" alt="approve_after" src="https://user-images.githubusercontent.com/1816888/162852686-1ad792ed-ec0a-474d-a960-5f8d3cd0e59f.png"> |






To test:

From Notifications > Comments list:
- Swipe left on a notification.
- Verify only the `Approve Comment`/`Unapprove Comment` button appears.
- Tap the button.
- Verify the events are now tracked:
`🔵 Tracked: notifications_comment_unapproved <blog_id: xxx, default_tab_experiment: nonexistent, site_type: p2, source: notifications>`
`🔵 Tracked: notifications_comment_approved <blog_id: xxx, default_tab_experiment: nonexistent, site_type: p2, source: notifications>`

From Notifications list:
- Swipe right on a notification.
- Select `Mark Unread`/`Mark Read`.
- Verify the events are now tracked:
`🔵 Tracked: notification_mark_as_unread_tapped <default_tab_experiment: nonexistent>`
`🔵 Tracked: notification_mark_as_read_tapped <default_tab_experiment: nonexistent>`

From Notifications > Comments list:
- Select a Comment notification.
- In the details, select `Pending` and `Approved`.
- Verify tracked events now include the `source`:
`🔵 Tracked: notifications_comment_unapproved <blog_id: xxx, default_tab_experiment: nonexistent, site_type: p2, source: notification_details>` 
`🔵 Tracked: notifications_comment_approved <blog_id: xxx, default_tab_experiment: nonexistent, site_type: p2, source: notification_details>`


## Regression Notes
1. Potential unintended areas of impact


2. What I did to test those areas of impact (or what existing automated tests I relied on)


3. What automated tests I added (or what prevented me from doing so)

PR submission checklist:
